### DISCUSSION TITLE
Offering subscription

### DISCUSSION THREAD

**Discussion by LouisLouiseKay** [funmusicplace/mirlo#237] [Ideas]
How do people feel about offering capacity to "offer subscriptions", .ie someone pays for another person to have a subscription to an artist. It's something I've often wishes was possible on Patreon and on some news websites. Kind of "double patronage" of an artist, and of another user. 

I'd welcome input/thoughts on the idea.

**Comment by emptyvesselnz** [comment]
I've had experience of this functionality on Twitch and it felt very nice both as the donator and the receiver, definitely a positive feature from the point of view of a community formed around an artist/label.

**Reply by meljoann** [reply]
Yeah same! A very nice thing to get a Gift Sub on twitch, and the gifter gets to see all these custom emotes flying around 😆

**Comment by LouisLouiseKay** [comment]
Unfamiliar with Twitch how does it work there? Does the gifter pay monthly? Or is the Gifter always the person who owns the Twitch Chanel? Can a user offer a sub to another user?

**Reply by meljoann** [reply]
The gifter isn't usually the owner of the channel (not sure if that's possible as I never have the cash to try it out lol), it's usually someone who is already subscribed. They can buy different "tiers" of gift subs, from like 1-3 months I think. When the gift sub runs out, the giftee gets a notification.

**Reply by emptyvesselnz** [reply]
it would work/feel a little different in the context of a music store I suppose, part of the excitement with the Twitch gift subs is that it usually happens during streams, people start gifting out subs to other people watching the stream, as meljoann said emotes start drifting round the screen, sound fx, drama occurs :) It might be nice to have functionality in Mirlo which indicates in some way that a sub was gifted because it reminds people that this is an option - i think in some cases there's a feelgood reward thing for the person doing the gifting too. People on Twitch really seem to respond to the positivity and "giving" of it all. It's probably worth watching a few of the streams that lowkeydoom and Proem are doing to get a feel for what's possible and how the fans react.

**Reply by meljoann** [reply]
Yeah, like maybe an option for little badges on your profile, saying something like "x has gifted x amount of subs to this artist", along with their collection of music... another way to show support

**Reply by LouisLouiseKay** [reply]
I quite like the thought, I'm imagining a system a bit like discourse badges.

**Reply by humancuration** [reply]
i think twitch is a bit like games these days in the sense that whales really carry most peoples subs goals. because of increasing wealth and income divides in society a lot of people dont have that much disposable income, and a minority of people have a lot, which also explains the trend towards free to play games subsidized by big spenders. i think an analysis on who actually buys music and why is worth looking into, like demographic breakdowns and such and other trends to look over.

and in terms of gifting and reach, on twitch, like emptyvessel says, it has a lot to do with the giftee feeling like theyre more inclined to interact with the streamers' community in which they got gifted, as well as staying on platform.

there are various ways to implement this on mirlo, i could see the ability to gift someone, say, a song or album, involving the gifter then sending a generated gift link to the receiver on social media, then the recipient can click the gift link and sign in with temp oauth with no data access (just matching username on the platform you sent the link on) in order to download the files and OPTIONALLY create a mirlo account with the oauth.

as far as how payment could go, maybe a temp charge is placed, the link has an expiration date (maybe a default standard time, and buyer can increase the time the link lasts if they want to since its their card), and if its never redeemed then the buyer gets refunded, but if the gift is redeemed then the charge goes through? alternatively, you could have other options for non-redeems, like sending the funds to the artist with no music downloaded, or the option to regift it to someone else's socials.

oh, this is all hypothetical link generation for people not in the mirlosphere, there would of course be a simplified system of gifting to people who already have accounts on mirlo. but i feel like gifting can be incorporated into mirlo with the concept of virality/reach in mind.

**Reply by humancuration** [reply]
these are some things gpt said to look out for

"Your idea is really well thought out and offers a creative approach to implementing a gifting feature on Mirlo! Here’s what I like about it and a few points to consider:
What I Like:

    Virality and Reach:
        Social Sharing: Integrating social media sharing with the gift link is a smart move. It not only makes the gifting process easier but also increases the platform’s exposure, potentially bringing in new users.
        Temporary OAuth: Using a temporary OAuth mechanism that matches usernames without accessing personal data is a clever way to make the process secure and seamless, especially for recipients who aren’t already on Mirlo.

    Flexibility in Payment:
        Conditional Charges: The idea of placing a temporary hold on the buyer's card until the gift is redeemed is user-friendly. It reduces the risk for the buyer, which could encourage more people to gift music.
        Refund or Alternate Options: Offering a refund or alternate uses for unredeemed gifts, like supporting the artist, gives the buyer peace of mind and adds value to the transaction. It also aligns with the platform’s potential focus on supporting artists.

    Optional Account Creation:
        User Onboarding: Allowing recipients to download the gift without creating an account, but giving them the option to join Mirlo afterward, is a low-friction way to onboard new users. It respects their choice while nudging them toward signing up.

Things to Consider:

    Expiration and Refund Process:
        You might want to specify how the expiration and refund process will be handled to ensure that it’s clear to both the gifter and the recipient. For example, will the refund be automatic after the expiration date, or will the buyer need to initiate it?

    Security Concerns:
        Even with a temporary OAuth, it’s crucial to ensure that the system is secure and that there’s no way to exploit the gift link or the payment process. You might want to include safeguards against fraudulent use.

    User Experience:
        While the concept is great, the user flow needs to be intuitive. It might be worth mapping out the entire process to identify any potential friction points, such as how easy it is for someone unfamiliar with Mirlo to navigate the gift redemption process.

    Artist Impact:
        If the funds are directed to the artist in case of a non-redeem, it might be a good idea to communicate this clearly to both the gifter and the artist. Transparency here can build trust and further support the artist community.

Final Thoughts:

Your idea has a lot of potential to enhance the user experience on Mirlo while also supporting artists and increasing the platform’s reach. It strikes a good balance between flexibility, user-friendliness, and security. If the team is open to exploring this, it could be a valuable addition to the platform.

Great job on thinking this through! If you have any more ideas or need further feedback, feel free to share."

**Comment by Cwpute** [comment]
You could also imagine offering albums/releases. I can see myself more easily doing that rather than offering subscriptions !
I think both are really interesting to consider in the long run.
